src/server/dflycmd.cc

```diff
@@ -366,6 +370,69 @@ void DflyCmd::ReplicaOffset(CmdArgList args, ConnectionContext* cntx) {
   }
 }
 
+void DflyCmd::ClusterManagmentCmd(CmdArgList args, ConnectionContext* cntx) {
+  if (!ClusterConfig::IsClusterEnabled()) {
+    return (*cntx)->SendError("DFLY CLUSTER commands requires --cluster_mode=yes");
+  }
+  // TODO check admin port
+  ToUpper(&args[1]);
+  string_view sub_cmd = ArgS(args, 1);
+  if (sub_cmd == "GETSLOTINFO") {
+    return ClusterGetSlotInfo(args, cntx);
+  }
+
+  return (*cntx)->SendError(UnknownSubCmd(sub_cmd, "DFLY CLUSTER"), kSyntaxErrType);
+}
+
+void DflyCmd::ClusterGetSlotInfo(CmdArgList args, ConnectionContext* cntx) {
+  if (args.size() == 3) {
+    return (*cntx)->SendError(facade::WrongNumArgsError("DFLY CLUSTER GETSLOTINFO"),
+                              kSyntaxErrType);
+  }
+  ToUpper(&args[2]);
+  string_view slots_str = ArgS(args, 2);
+  if (slots_str != "SLOTS") {
+    return (*cntx)->SendError(kSyntaxErr, kSyntaxErrType);
+  }
+
+  vector<std::pair<SlotId, SlotStats>> slots_stats;
+  for (size_t i = 3; i < args.size(); ++i) {
+    string_view slot_str = ArgS(args, i);
+    uint32_t sid;
+    if (!absl::SimpleAtoi(slot_str, &sid)) {
+      return (*cntx)->SendError(kInvalidIntErr);
+    }
+    if (sid > ClusterConfig::kMaxSlotNum) {
+      return (*cntx)->SendError("Invalid slot id");
+    }
+    slots_stats.push_back(make_pair(sid, SlotStats{}));
+  }
+
+  Mutex mu;
+
+  auto cb = [&](auto*) {
+    EngineShard* shard = EngineShard::tlocal();
+    if (shard == nullptr)
+      return;
+
+    lock_guard lk(mu);
+    for (auto& slot_data : slots_stats) {
+      slot_data.second += shard->db_slice().GetSlotStats(slot_data.first);
```

Comment:
If you wish, you could do `for (auto& [slot, data] : slots_stats)` and then drop the `.first` and `.second` below.
That should work despite the container not being a map.